src/porepy/numerics/ad/operators.py

```diff
@@ -190,72 +287,25 @@ def set_name(self, name: str) -> None:
         """
         self._name = name
 
-    def previous_timestep(self) -> pp.ad.Operator:
```

Comment:
Base class has now a prev iter and time step method which invokes the recursion.